src/test/func/redblack/redblack.cc

```diff
@@ -21,16 +21,42 @@ struct Wrapper
   // the representation.
   static constexpr size_t offset = 10000;
 
-  size_t value = offset << 1;
+  size_t* ptr;
+  constexpr Wrapper(size_t* p) : ptr(p) {}
```

Comment:
Changed to `NodeRef`.  I've also extended the test so that the internal storage is `size_t` but the interface is in terms of `uint16_t`, so that we test both using a wrapper class for accessing and having a different internal representation and interface type.